dymos/phase/phase.py

```diff
@@ -1003,19 +1004,32 @@ def add_boundary_constraint(self, name, loc, constraint_name=None, units=None,
             Value of response variable that scales to 0.0 in the driver.
         linear : bool
             Set to True if constraint is linear. Default is False.
+        single_var : bool
```

Comment:
I don't think you need this `single_var` argument in `add_boundary_constraint`. Dymos should be able to tell if it is an expression or just a variable. If it has any of these characters: ['(', '+', '-', '/', '*'] in it, then it is definitely an expression. If it does not, it is definitely a variable.  (Note: I didn't include modulus "%" or bitwise "&" because I don't think we would use those in an expression, but for completeness, you could include those in the check too.)

This will change some of the logic in other parts of the code, so I won't comment on that yet.